src/query/generated/proto/rpcpb/query.proto

```diff
@@ -90,3 +92,44 @@ message M3Segment {
 	int64 startTime = 3;
 	int64 blockSize = 4;
 }
+
+message SearchRequest {
+	oneof matchers {
+		TagMatchers tagMatchers     = 1;
+	}
+	SearchType type               = 2;
+	repeated bytes filterNameTags = 3;
+}
+
+enum SearchType {
+	// Requests tag names and values
+	DEFAULT = 0;
+	// Requests tag names only
+	TAGNAME = 1;
+}
+
+message SearchResponse {
+	oneof value {
+		M3TagProperties compressed = 1;
+		TagProperties decompressed = 2;
```

Comment:
Hey, I just realized that it's a little more complicated that we first thought.  There's actually two APIs that we need.

There's basically two types of queries:
1. Find series by tags, similar to Prometheus "Find series by label matchers":
https://prometheus.io/docs/prometheus/latest/querying/api/#finding-series-by-label-matchers
2. Find tag values, similar to Prometheus "Querying label values"
https://prometheus.io/docs/prometheus/latest/querying/api/#querying-label-values

You can see for the second API, on our current query service we actually flatten the series together (you don't return any "per series" results):
```go
func renderSearchQueryResultsJSON(
	w http.ResponseWriter,
	result *index.MetadataTags,
) error {
	jw := streamjson.NewWriter(w)
	jw.BeginObject()

	// ...

	jw.BeginObjectField("tags")
	jw.BeginArray()
	// Sort for tests and deterministic output
	index.SortTagProperties(result.Tags)
	for _, tag := range result.Tags {
		jw.BeginObject()

		jw.BeginObjectField("key")
		jw.WriteString(tag.Key)

		// ...

		jw.BeginObjectField("values")

		jw.BeginArray()
		// Sort for tests and deterministic output
		sort.Strings(tag.Values)
		for _, value := range tag.Values {
			jw.WriteString(value)
		}
		jw.EndArray()

		jw.EndObject()
	}
	jw.EndArray()

	jw.EndObject()

	return jw.Close()
}
```

Reply:
Ah gotcha, yeah I was trying to bridge between the current search endpoints we had in coordinator/query and the existing remote endpoint we have.

If I'm understanding correctly, you're saying that this PR satisfies what you described as query type 1, right? If it does neither I'll be a very sad boy haha